### PR TITLE
SW-8376 Delete observation data for published activities

### PR DESCRIPTION
**Pull request by sgrimm**
When a published activity is deleted, we need to delete any observation-specific
details about it and its media files.

### CODE REVIEW

**Comment by sgrimm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/terraware/terraware-server/4444?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4460** <a href="https://app.graphite.com/github/pr/terraware/terraware-server/4460?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4445** <a href="https://app.graphite.com/github/pr/terraware/terraware-server/4445?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4444** <a href="https://app.graphite.com/github/pr/terraware/terraware-server/4444?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/terraware/terraware-server/4444?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#4443** <a href="https://app.graphite.com/github/pr/terraware/terraware-server/4443?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/terraware?ref=merge-queue-instructions-in-stack-comment&prId=4444).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/test/kotlin/com/terraformation/backend/funder/PublishedActivityServiceTest.kt [hunk 1]
```diff
@@ -81,10 +83,18 @@ class PublishedActivityServiceTest : DatabaseTest(), RunsAsDatabaseUser {
...
+      insertSubstratum()
+      insertMonitoringPlot()
+      insertObservation()
+      insertActivityObservation()
+
       activityFileId = insertFile()
       insertActivityMediaFile()
       insertPublishedActivity()
       insertPublishedActivityMediaFile()
+      insertPublishedActivityObservationMediaFile()
```
Comment (ice1080) [inline src/test/kotlin/com/terraformation/backend/funder/PublishedActivityServiceTest.kt:98]: Don't we also want to test that the `PUBLISHED_ACTIVITY_OBSERVATION_MEDIA_FILES` is empty after deleting?

Comment (sgrimm) [reply src/test/kotlin/com/terraformation/backend/funder/PublishedActivityServiceTest.kt:98]: The foreign key constraint should make it impossible for that table to have files that aren't in `PUBLISHED_ACTIVITY_MEDIA_FILES`, but good call, better to be explicit about the expected behavior.

Comment (ice1080) [reply src/test/kotlin/com/terraformation/backend/funder/PublishedActivityServiceTest.kt:98]: Ya we still usually write unit tests for constraints like that which is why I commented. Thanks Steve!

**Review by ice1080 — APPROVED** [review]

**Comment by sgrimm** [comment]
### Merge activity

* **May 26, 10:40 PM UTC**: The merge label 'merge-queue' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=terraware&repo=terraware-server) once it meets the requirements.
* **May 26, 10:56 PM UTC**: `sgrimm` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=terraware&repo=terraware-server).
* **May 26, 10:57 PM UTC**: CI is running for this pull request on a draft pull request ([#4461](https://app.graphite.com/github/pr/terraware/terraware-server/4461)) due to your merge queue CI optimization settings.
* **May 26, 11:18 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=terraware&repo=terraware-server) via draft PR: [#4461](https://app.graphite.com/github/pr/terraware/terraware-server/4461).
